chore: improve documentation & examples (#1061)
graphql/graphiql: README.md

[![NPM](https://img.shields.io/npm/v/graphiql.svg)](https://npmjs.com/graphiql)
[![CDNJS](https://img.shields.io/cdnjs/v/graphiql.svg)](https://cdnjs.com/libraries/graphiql)

_/ˈɡrafək(ə)l/_ A graphical interactive in-browser GraphQL IDE. [Try the live demo](http://graphql.org/swapi-graphql). We also have [a demo using our latest netlify build](http://graphiql-test.netlify.com) for the master branch.

The GraphiQL IDE, implemented in React, currently using [GraphQL mode for CodeMirror](packages/codemirror-graphql#readme) & [GraphQL Language Service](packages/graphql-language-service#readme).

### How To Setup/Implement GraphiQL
[![Edit graphiql-example](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/graphiql-example-nhzvc)

> (requires no create react app configuration)

Reviewer: Make this a little clearer:

```suggestion
(works with `create-react-app` without requiring additional configuration)
```

[CONTRIBUTING.md](CONTRIBUTING.md) for details on how to contribute.

## Developing/Contributing

This repo is a yarn workspaces monorepo that also uses lerna for some convenience.

> It requires node 11 and the latest yarn ideally.  `npm` commands _will_ fail.

Reviewer: ```suggestion
It requires node 11 and the latest yarn ideally. Running these commands with `npm` _will_ cause you problems.
```

> As of December 2019 we are officially supporting windows OS for development tooling. If you encounter any bugs when using these or other npm scripts, pleas report them!

Reviewer: ```suggestion
As of December 2019 we are officially supporting Windows OS for development tooling. If you encounter any bugs when using these or other package.json scripts, please report them!
```

### Getting Started

1. `yarn` - install and link all packages
2. `yarn build` - cleans first, then builds everything but webpack bundles - `tsc --build`, `babel` etc
3. `yarn build-bundles` - builds webpack bundles that are used for releases

> 4. `yarn build-demo` - builds demo projects for netlify. we run this on CI to make sure webpack can consume our project in a standalone project.

Reviewer: ```suggestion
4. `yarn build-demo` - builds demo projects for netlify; we run this on CI to make sure webpack can consume our project in a standalone project.
```

5. `yarn test` - runs all of the above alongside linting and checks, jest mocha Cypress etc.
6. `yarn pretty` - autoformats
7. `yarn lint` - checks for linting issues

> 8. `yarn e2e` - just runs cypress headless, ala CI against minified bundle and a local schema server.

Reviewer: ```suggestion
8. `yarn e2e` - runs cypress headlessly against the minified bundle and a local schema server, like in CI.
```

> 9. `yarn jest` - runs global jest commands across the entire monorepo. try `yarn jest --watch` or `yarn jest DocExplorer` for example :D

Reviewer: ```suggestion
9. `yarn jest` - runs global jest commands across the entire monorepo; try `yarn jest --watch` or `yarn jest DocExplorer` for example :D
```
